tiledb/api/c_api/subarray/subarray_api_experimental.h

```diff
@@ -66,6 +66,40 @@ TILEDB_EXPORT capi_return_t tiledb_subarray_add_point_ranges(
     const void* start,
     uint64_t count) TILEDB_NOEXCEPT;
 
+/**
+ * Adds variable-sized point ranges to the given dimension index of the
+ * subarray. Effectively `add_range(x_i, x_i)` for all points in the
+ * target array, but set in bulk to amortize expensive steps.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * const char* buffer = "TileDBtest";
+ * uint64_t buffer_size = 10;
+ * uint64_t offsets[] = {0, 2, 3, 6, 8};
+ * uint64_t offsets_size = 5;
+ * tiledb_subarray_add_point_ranges_var(ctx, subarray, 0, buffer,
+ * buffer_size, offsets, offsets_size);
+ * @endcode
```

Comment:
Nit : I'd prefer if we also updated the example here with "aabb..."? "TileDBtest" seems like array data to me and it got me confused in the beginning, I didn't realize those were concatenated ranges.